projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-notification-watcher.ts

```diff
@@ -0,0 +1,123 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { useContext, useEffect } from 'react';
+import { MyJetpackRoutes } from '../../constants';
+import { NOTICE_PRIORITY_MEDIUM, NOTICE_PRIORITY_HIGH } from '../../context/constants';
+import { NoticeContext } from '../../context/notices/noticeContext';
+import { useAllProducts } from '../../data/products/use-product';
+import { getMyJetpackWindowInitialState } from '../../data/utils/get-my-jetpack-window-state';
+import getProductSlugsThatRequireUserConnection from '../../data/utils/get-product-slugs-that-require-user-connection';
+import useAnalytics from '../use-analytics';
+import useMyJetpackNavigate from '../use-my-jetpack-navigate';
+
+type RedBubbleAlerts = Window[ 'myJetpackInitialState' ][ 'redBubbleAlerts' ];
+
+const useNotificationWatcher = () => {
+	const { redBubbleAlerts } = getMyJetpackWindowInitialState();
+
+	useBadInstallNotice( redBubbleAlerts );
+	useSiteConnectionNotice( redBubbleAlerts );
+};
+
+export default useNotificationWatcher;
+
+const useSiteConnectionNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
```

Comment:
In the long term, I think we can move these functions to separate files to make it more maintainable/readable